src/main/java/com/google/sps/data/User.java

```diff
@@ -208,4 +207,26 @@ public String toJSON() {
       String json = gson.toJson(this);
       return json;
   }
+
+  /* Function embeds Map of challenge_statuses into an Entity so it may
+  be stored in Datastore */
+  private EmbeddedEntity embedChallengeStatuses(){
```

Comment:
[No action required]: An alternate way to do this is to serialize the entire map to JSON and write the JSON into datastore.

Reply:
I like that idea. thanks for the feedback